### ISSUE TITLE
How do we expect people to use hooks?

### ISSUE THREAD

**Issue by bmw** [certbot/certbot#4564]
I think pre, post, and renew hooks in Certbot are quite complicated and hard to use. There's a number of rules for when each hooks is run, what options overwrite previously defined hooks, and options for users where to define their hooks:

* renewal configuration files
* global configuration file
* command line
* `*-hook.d` directories in the Ubuntu PPA

We should try to come up with a clear picture for what we expect users to do here and guide them through the process. I think we should also try to simplify things where possible without breaking existing behavior.

**Comment by SwartzCr** [comment]
The *-hook.d directories seem to totally break functionality for Ubuntu PPA users.
For example if I had a system packaged version of certbot (or letsencrypt) and I specified hooks, then I upgraded to a packaged version, now the hooks specified in my renewal files no longer work. This probably means that my renewals are broken, which is not something that we should ever do.
Also none of this is documented, which makes me really sad
